main/acle.md

```diff
@@ -2699,11 +2704,41 @@ following:
   in one of the translation units.
   * Implicitly, without this attribute,
   * or explicitly providing the `default` in the attribute.
-* All instances of the versions shall share the same function
-  signature and calling convention.
+* The default version signature is the signature for calling
+  the multiversioned functions.
+* Non-default versions shall share the same calling convention
+  of the default version and have a type that is convertable to the type
+  of the default version.
+
 * All the function versions must be declared at the translation
   unit in which the definition of the default version resides.
 
+For example, in the case below:
+```C
+__attribute__((target_version("default"))) void fn(void);
+__attribute__((target_version("foo"))) void fn(void); // f_foo below
+````
+
+This case is valid as it is possible to use the
+non-default function as the value for a pointer with the type of
+the default function, as shown below.
```

Comment:
Perhaps you are mistaken here? We cannot explicitly refer to a specific version. Taking the address of a versioned function and assigning it into a pointer will take the address of the ifunc symbol.

Reply:
Agreed. That said, I'd like to have syntax for that: https://github.com/llvm/llvm-project/issues/84094

Reply:
This example is intended to clarify what is meant by `Non-default versions shall [..] have a type that is convertible to the type of the default version`. 
I intended to show that a non-default function has "convertible" signature if a function of the non-default signature could be assigned to a function pointer with the default functions signature.

This was a suggestion by Richard Sandiford, that we should use existing function pointer rules to define signature/type compatibility. However, I struggled to formalize it and clearly this example has is not clear for you two, let alone non-experts, so unless you have a way to write this clearer, I will remove it.